libs/ui/src/hooks/auth/use_dipped_smart_card_auth.ts

```diff
@@ -0,0 +1,44 @@
+import useInterval from 'use-interval';
+import { DippedSmartCardAuth } from '@votingworks/types';
+import { useEffect, useState } from 'react';
+
+const AUTH_STATUS_POLLING_INTERVAL = 100;
+
+export interface AuthApiClient {
+  getAuthStatus: () => Promise<DippedSmartCardAuth.AuthStatus>;
+  logOut: () => Promise<void>;
+}
+
+export function useDippedSmartCardAuth(
+  authApiClient: AuthApiClient
+): DippedSmartCardAuth.AuthStatus {
+  const [authStatus, setAuthStatus] = useState<DippedSmartCardAuth.AuthStatus>(
+    DippedSmartCardAuth.DefaultAuthStatus
+  );
+  const [hasBeenReset, setHasBeenReset] = useState(false);
+
+  useEffect(() => {
+    async function logOut() {
+      await authApiClient.logOut();
+      setHasBeenReset(true);
+    }
+    void logOut();
+    // Log out once on mount to avoid inconsistencies between the backend and the frontend when the
+    // frontend restarts and the backend does not (e.g. during frontend development)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I was initially super excited about the fact that we'd no longer have to auth over and over every time the page refreshed during frontend development since auth state is now managed on the backend and only cleared when the backend restarts.

But I ultimately ran into various (development-specific) errors when the frontend restarted but the backend didn't. Here's an example that actually came up in Cypress integration tests. If you auth as an election manager (and don't log out) and then open an incognito window to simulate Cypress starting a new test in a new browser, you'll hit an assertion error [here](https://github.com/votingworks/vxsuite/blob/77bd5851ed42380c7937aad34f022ab908354ab7/frontends/election-manager/src/screens/ballot_list_screen.tsx#L41) because you're still authenticated on the backend but an election definition hasn't been defined yet in the new browser.

Still thinking through a few more edge cases but I've thus far found logging out on mount to be a simple way to keep things in sync

Reply:
Further thinking on this below in https://github.com/votingworks/vxsuite/pull/2960#discussion_r1092700857